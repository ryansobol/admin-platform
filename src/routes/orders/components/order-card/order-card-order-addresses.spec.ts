import { render, screen } from '@testing-library/svelte';
import { expect, it } from 'vitest';

import OrderCardOrderAddresses from './order-card-order-addresses.svelte';
import type { Order } from '../../types';

it('renders nothing without an order prop', () => {
	render(OrderCardOrderAddresses);

	const element = screen.queryByRole('region', { name: 'Order Addresses' });

	expect(element).toBeNull();
});

it('renders the component with an order prop that has both shipping and billing addresses', () => {
	const order: Order = {
		id: 1,
		code: 'ABC123',
		customer: {
			name: 'Firstname Lastname',
			email: 'firstname.lastname@example.com',
			phone: '+1-234-567-8900'
		},
		type: 'Monthly Subscription',
		status: 'Fulfilled',
		createdAt: '2023-01-09T12:11:27-07:00',
		updatedAt: '2023-01-11T04:11:27-07:00',
		products: [{ id: 7, name: 'Handmade Ceramic Mug', quantity: 4, price: 304 }],
		subtotal: 1216,
		shipping: 26.99,
		tax: 109.44,
		total: 1352.43,
		shippingAddress: {
			line1: 'Firstname Lastname',
			line2: '117 1st Ave.',
			line3: 'Phoenix, NY 60975'
		},
		billingAddress: {
			line1: 'Firstname Lastname',
			line2: '117 1st Ave.',
			line3: 'Phoenix, NY 60975'
		},
		creditCard: {
			type: 'Visa',
			numberRedacted: '**** **** **** 3088'
		}
	};

	render(OrderCardOrderAddresses, { order });

	const element = screen.getByRole('region', { name: 'Order Addresses' });

	expect(element).toMatchSnapshot();
});

it('renders the component with an order prop that has a shipping address but no billing address', () => {
	const order: Order = {
		id: 1,
		code: 'ABC123',
		customer: {
			name: 'Firstname Lastname',
			email: 'firstname.lastname@example.com',
			phone: '+1-234-567-8900'
		},
		type: 'Monthly Subscription',
		status: 'Fulfilled',
		createdAt: '2023-01-09T12:11:27-07:00',
		updatedAt: '2023-01-11T04:11:27-07:00',
		products: [{ id: 7, name: 'Handmade Ceramic Mug', quantity: 4, price: 304 }],
		subtotal: 1216,
		shipping: 26.99,
		tax: 109.44,
		total: 1352.43,
		shippingAddress: {
			line1: 'Firstname Lastname',
			line2: '117 1st Ave.',
			line3: 'Phoenix, NY 60975'
		},
		billingAddress: null,
		creditCard: {
			type: 'Visa',
			numberRedacted: '**** **** **** 3088'
		}
	};

	render(OrderCardOrderAddresses, { order });

	const element = screen.getByRole('region', { name: 'Order Addresses' });

	expect(element).toMatchSnapshot();
});
