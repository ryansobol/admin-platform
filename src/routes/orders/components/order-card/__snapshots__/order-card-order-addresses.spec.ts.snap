// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the component with an order prop that has a shipping address but no billing address 1`] = `
<section
  aria-label="Order Addresses"
  class="grid grid-cols-2 gap-4"
>
  <section
    aria-labelledby="shipping-address"
    class="grid gap-3"
  >
    <h4
      class="font-semibold"
      id="shipping-address"
    >
      Shipping Address
    </h4>
     
    <address
      class="grid gap-0.5 not-italic text-muted-foreground"
    >
      <span>
        Firstname Lastname
      </span>
       
      <span>
        117 1st Ave.
      </span>
       
      <span>
        Phoenix, NY 60975
      </span>
    </address>
  </section>
   
  <section
    aria-labelledby="billing-address"
    class="grid auto-rows-max gap-3"
  >
    <h4
      class="font-semibold"
      id="billing-address"
    >
      Billing Address
    </h4>
     
    <div
      class="text-muted-foreground"
    >
      Same as shipping address
    </div>
    <!---->
  </section>
</section>
`;

exports[`renders the component with an order prop that has both shipping and billing addresses 1`] = `
<section
  aria-label="Order Addresses"
  class="grid grid-cols-2 gap-4"
>
  <section
    aria-labelledby="shipping-address"
    class="grid gap-3"
  >
    <h4
      class="font-semibold"
      id="shipping-address"
    >
      Shipping Address
    </h4>
     
    <address
      class="grid gap-0.5 not-italic text-muted-foreground"
    >
      <span>
        Firstname Lastname
      </span>
       
      <span>
        117 1st Ave.
      </span>
       
      <span>
        Phoenix, NY 60975
      </span>
    </address>
  </section>
   
  <section
    aria-labelledby="billing-address"
    class="grid auto-rows-max gap-3"
  >
    <h4
      class="font-semibold"
      id="billing-address"
    >
      Billing Address
    </h4>
     
    <address
      class="grid gap-0.5 not-italic text-muted-foreground"
    >
      <span>
        Firstname Lastname
      </span>
       
      <span>
        117 1st Ave.
      </span>
       
      <span>
        Phoenix, NY 60975
      </span>
    </address>
    <!---->
  </section>
</section>
`;
